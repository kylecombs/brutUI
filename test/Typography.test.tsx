import * as React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import { Typography } from "../src";

describe("Typography", () => {
  it("Renders correctly", () => {
    const { asFragment } = render(
      <Typography variant="h1">Heading example</Typography>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("Check if renders correct element (h1)", () => {
    const content = "Heading example";

    const { getByText } = render(
      <Typography variant="h1">{content}</Typography>,
    );

    const textElement = getByText(content);
    expect(textElement.outerHTML).toMatch("h1");
  });

  it("Check if renders correct element (label)", () => {
    const content = "Heading example";

    const { getByText } = render(
      <Typography as="label" variant="h1">
        {content}
      </Typography>,
    );

    const textElement = getByText(content);
    expect(textElement.outerHTML).toMatch("label");
  });

  it("Check if renders correct element (p)", () => {
    const content = "Paragraph example";

    const { getByText } = render(
      <Typography variant="xl">{content}</Typography>,
    );

    const textElement = getByText(content);
    expect(textElement.outerHTML).toMatch("p");
  });

  it("Should have the correct h1 ", () => {
    const content = "Heading example";

    const { getByText } = render(
      <Typography variant="h1">{content}</Typography>,
    );

    const textElement = getByText(content);
    expect(textElement.classList).toContain("text-h1");
  });

  it("Use custom color", () => {
    const content = "Heading example";
    const customColor = "text-primary-600 dark:text-gray-300";

    const { getByText } = render(
      <Typography variant="h1" customColor={customColor}>
        {content}
      </Typography>,
    );

    const textElement = getByText(content);
    expect(textElement.classList).toContain("text-primary-600");
    expect(textElement.classList).toContain("dark:text-gray-300");
  });

  it("Use custom weight", () => {
    const content = "Heading example";

    const { getByText } = render(
      <Typography variant="h1" customWeight="medium">
        {content}
      </Typography>,
    );

    const textElement = getByText(content);
    expect(textElement.classList).toContain("font-medium");
    expect(textElement.classList.contains("font-regular")).toBe(false);
  });
});
