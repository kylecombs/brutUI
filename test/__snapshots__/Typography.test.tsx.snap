// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography Renders correctly 1`] = `
<DocumentFragment>
  <h1
    class="text-h1 font-normal tracking-tight text-black dark:text-white"
  >
    Heading example
  </h1>
</DocumentFragment>
`;
